sklearn/preprocessing/_discretization.py

```diff
@@ -136,6 +143,13 @@ def fit(self, X, y=None):
         self
         """
         X = check_array(X, dtype='numeric')
+        n_samples, n_features = X.shape
+
+        if n_samples > self.subsample:
```

Comment:
I believe we should also allow for `subsample=None` in which case subsampling is disabled.

Reply:
We should also add that option to `QuantileTransformer` but in a separate PR.

Reply:
Done